spec/controllers/spree/admin/orders_controller_spec.rb

```diff
@@ -13,7 +13,7 @@
     it "advances the order state" do
       expect {
         spree_get :edit, id: order
-      }.to change { order.reload.state }.from("cart").to("payment")
+      }.to change { order.reload.state }.from("cart").to("complete")
```

Comment:
This "event" where an admin order goes from being essentially a _draft_ to being a competed order is quite significant, in that it reduces the actual stock of all the items in the order. What's missing from the flow here is any kind of feedback to the user that indicates what is happening at these different points. It feels like it needs some thinking!

This transition is handled in the `Spree::Admin::OrdersController#edit` action right? There's this line (29) where we blindly try to kick the order through as many states as it can (towards completion):

https://github.com/openfoodfoundation/openfoodnetwork/blob/088ae496cc8a0f7b438887246e8fe2cd42bfcc08/app/controllers/spree/admin/orders_controller.rb#L26-L30


The fact that we're making serious changes to an object inside of an `edit` action is already pretty alarming. I think we probably should not be doing this...

Anyway, what I'm getting at is: I'm wondering if we should have some kind of step or explicit interaction from the user before we set the order to `complete`. Like clicking a button that says "Place Order" or something, and having it convey that the order will be competed and the stock will be reduced during that action.

Reply:
> where we blindly try to kick the order through as many states as it can (towards completion) 

😂 

Yeah this tripped me up when I first read it. 

I think we can finish this PR and have it be an improvement without a total rework of the method... but this would be something that might fit in nicely with the order states work that's already underway.